src/pages/index.js

```diff
@@ -8,62 +8,204 @@ import {
   selectors,
   profileTitleInput,
   profileDescriptionInput,
+  avatarModalFormSelector,
+  editButtonAvatar,
+  avatarSelector,
+  cardDeleteModal,
 } from "../utils/constants.js";
 import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  authToken: "cb447498-fb2c-4c99-9fc5-8ee58bc7fe4c",
+});
 
 // Popup with Image
 const cardPreview = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 cardPreview.setEventListeners();
 
-// Section / Card
-
+// Section / Card Create Card
 const renderCard = (data) => {
+  const likes = data.likes || [];
   const cardElement = new Card(
     {
       data,
       handleImageClick: (imageData) => {
         cardPreview.open(imageData);
       },
+      handleDeleteClick: () => {
+        deleteModal.open();
+        deleteModal.setSubmitAction(() => {
+        deleteModal.renderLoading(true);
+        const id = cardElement.getId();
+        api
+          .deleteCard(id)
+          .then(() => {
+          cardElement.handleDeleteButton();
+          deleteModal.close();
+        })
+        .catch(console.error)
+        .finally(() => {
+          deleteModal.renderLoading(false);
+        });
+      });
+      },
+
+      handleLikeClick: () => {
+        const id = cardElement.getId();
+        console.log();
+        if (cardElement.isLiked()) {
+          api
+            .unLikeCard(id)
+            .then((data) => {
+              cardElement.setLikes(data.likes);
+            })
+            .catch((err) => console.error(err));
+        } else {
+          api
+            .likeCard(id)
+            .then((data) => {
+              cardElement.setLikes(data.likes);
+            })
+            .catch((err) => console.error(err));
+        }
+      },
+      myId: userId,
     },
     selectors.cardTemplate
   );
+
   const newCard = cardElement.getView();
   cardSection.addItem(newCard);
 };
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+const deleteModal = new PopupWithForm({
+  handleFormSubmit: () => {
+    deleteModal.renderLoading(true);
   },
-  selectors.cardSection
-);
-cardSection.renderItems();
+  popupSelector: cardDeleteModal,
+  loadingText: "Deleting...",
+});
 
 // User Info
-const user = new UserInfo(".profile__title", ".profile__subtitle");
+const user = new UserInfo(
+  ".profile__title",
+  ".profile__subtitle",
+  ".profile__image"
+);
+
+//pulling in the card data via API
+let cardSection;
+let userId;
+
+Promise.all([api.getUserInfo(), api.getCardList()])
+  .then(([userData, data]) => {
+    user.setUserInfo({
+      title: userData.name,
+      subtitle: userData.about,
+      avatar: userData.avatar,
```

Comment:
Please change the avatar only via `setAvatarInfo` method

Reply:
Done